Jint/Native/Array/ArrayInstance.cs

```diff
@@ -20,7 +21,7 @@ public class ArrayInstance : ObjectInstance, IEnumerable<JsValue>
 
         private PropertyDescriptor _length;
 
-        private const int MaxDenseArrayLength = 1024 * 10;
+        public const int MaxDenseArrayLength = 1024 * 10;
```

Comment:
Why is this public?

Reply:
We need to know whether we want to send and array or a dictionary to the ctor.

Reply:
Doe Raven ever actually need to jump from dense to sparse? Dictionary is mostly for JS madness where you can have indexes 0 and 1000000 without nothing in between (growth algorithm tries to detect over 50 index jumps and goes to sparse then..

Reply:
what is the source here? We are fine with having a dense array that is 50,000 items long

Reply:
First, I sincerely hope that there will not be documents with 10K long arrays :) About the dictionary, I don't think it's only about Raven.. The criteria in ArrayInstance whether to use a dictionary or an array is the collection size, which means that once your "capacity" is bigger then 10k, we are going to pay for Dictionary.Insert, which is _very_ expensive... On the other side, I understand the scenario with non-sequential arrays. I'll change the code to support array only or dictionary only ctors, while the question about how to differ between long sequential arrays and non-sequential arrays will stay open, and should be addressed in the "original" ctor